pkg/cmd/cli/cmd/cancelbuild.go

```diff
@@ -3,166 +3,234 @@ package cmd
 import (
 	"fmt"
 	"io"
+	"strings"
+	"sync"
+	"time"
 
-	"github.com/golang/glog"
 	"github.com/spf13/cobra"
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/errors"
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
-	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/wait"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	osclient "github.com/openshift/origin/pkg/client"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 )
 
 const (
 	cancelBuildLong = `
-Cancels a pending or running build
+Cancels a pending, running or new builds
 
-This command requests a graceful shutdown of the running build. There may be a delay between requesting 
+This command requests a graceful shutdown of the build. There may be a delay between requesting 
 the build and the time the build is terminated.`
 
 	cancelBuildExample = `  # Cancel the build with the given name
-  $ %[1]s cancel-build 1da32cvq
+  $ %[1]s cancel-build ruby-build-2
 
   # Cancel the named build and print the build logs
-  $ %[1]s cancel-build 1da32cvq --dump-logs
+  $ %[1]s cancel-build ruby-build-2 --dump-logs
 
   # Cancel the named build and create a new one with the same parameters
-  $ %[1]s cancel-build 1da32cvq --restart`
+  $ %[1]s cancel-build ruby-build-2 --restart
+
+  # Cancel multiple builds
+  $ %[1]s cancel-build ruby-build-1 ruby-build-2 ruby-build-3
+
+  # Cancel all builds created from 'ruby-build' build configuration that are in 'new' state
+  $ %[1]s cancel-build bc/ruby-build --state=new`
 )
 
+type CancelBuildOptions struct {
+	In          io.Reader
+	Out, ErrOut io.Writer
+
+	DumpLogs   bool
+	Restart    bool
+	State      string
+	Namespace  string
+	BuildNames []string
+
+	Client      osclient.Interface
+	BuildClient osclient.BuildInterface
+}
+
 // NewCmdCancelBuild implements the OpenShift cli cancel-build command
-func NewCmdCancelBuild(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+func NewCmdCancelBuild(fullName string, f *clientcmd.Factory, in io.Reader, out io.Writer) *cobra.Command {
+	o := &CancelBuildOptions{}
+
 	cmd := &cobra.Command{
-		Use:        "cancel-build BUILD",
-		Short:      "Cancel a pending or running build",
+		Use:        "cancel-build (BUILD | BUILDCONFIG)",
+		Short:      "Cancel a pending, running or new builds",
```

Comment:
We use the Oxford comma on this team.  I have to remember to put that in
the Kube dev conventions.

On Fri, Apr 15, 2016 at 3:14 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/cancelbuild.go
> <https://github.com/openshift/origin/pull/8509#discussion_r59926572>:
>
> >  	cmd := &cobra.Command{
> > -		Use:        "cancel-build BUILD",
> > -		Short:      "Cancel a pending or running build",
> > +		Use:        "cancel-build (BUILD | BUILDCONFIG)",
> > +		Short:      "Cancel a pending, running or new builds",
>
> "Cancel a pending, running, or new build"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8509/files/35e294738e5de6039065e027a7e4157ceefb9792#r59926572>
>